cmd/habitat/community/consensus/raft/helpers.go

```diff
@@ -0,0 +1,53 @@
+package raft
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/eagraf/habitat/pkg/compass"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	Localhost  = "localhost"
+	DockerHost = "0.0.0.0"
+	Protocol   = "http"
+
+	MultiplexerPort = "6000"
+	RPCPort         = "6001"
+)
+
+func getMultiplexerAddress() string {
+	ip, err := compass.LocalIPv4()
+	if err != nil {
+		log.Fatal().Msgf("error getting local IP address: %s", err)
+	}
+
+	return fmt.Sprintf("%s:%s", ip.String(), MultiplexerPort)
+}
+
+func getRPCAddress() string {
+	host := Localhost
+	if compass.InDockerContainer() {
+		host = compass.Hostname()
+	}
+
+	return fmt.Sprintf("%s:%s", host, RPCPort)
+}
+
+func getServerID(communityID string) string {
+	nodeID := compass.NodeID()
```

Comment:
wait, can you add a comment somewhere explaining what node id is? I'm confused if it refers to the ipfs node, raft node, or something that we are defining ourselves

Reply:
When habitat starts for the first time, I write a new UUID to $HABITAT_PATH/node_id so that we can get a unique identifier for the habitat instance that is stable across restarts

Reply:
The probability of UUID collisions is extremely small (like even if everyone on earth used habitat, it probably would not happen). I don't know if its possible to guarantee uniqueness without some sort of centralized service, but I think that its likely ok to skip on this for now